### ISSUE TITLE
Feature Request: Hooks support.

### ISSUE THREAD

**Issue by Svenum** [AshleyYakeley/NixVirt#27]
I would love to add hooks to the VMs via this module. I know its already possible https://search.nixos.org/options?channel=unstable&show=virtualisation.libvirtd.hooks&from=0&size=50&sort=relevance&type=packages&query=hooks but it would be nice to do this also over this module.

**Comment by AshleyYakeley** [comment]
What kind of hooks? Could you give me an example of what you want to do?

**Comment by Svenum** [comment]
I mean this hooks here: https://www.libvirt.org/hooks.html

I want to add some scripts before the VM starts and after it is shutdown.

**Comment by AshleyYakeley** [comment]
The libvirt hooks are already available in NixOS: https://search.nixos.org/options?channel=23.11&from=0&size=50&sort=relevance&type=packages&query=libvirtd.hooks

Alternatively I can add hooks for when NixVirt does activation/deactivation of domains (and other objects).

**Comment by AshleyYakeley** [comment]
Will the NixOS libvirtd hooks work for you or should I add something?

**Comment by Svenum** [comment]
Yes it would be nice if you add the support for hooks. For example the default module form Nix could not create supfolders in the hook directory. That is usefull If you dont want to write a complex script with much if else conditions, only to check if the host and the vm stat is right. (https://passthroughpo.st/simple-per-vm-libvirt-hooks-with-the-vfio-tools-hook-helper/). And maybe its possible to write hooks for user sessions (qemu:///session), but I do not know.

**Comment by AshleyYakeley** [comment]
OK, if I did this it would add per-domain hooks via NixOS `virtualisation.libvirtd.hooks.*` and therefore only work with NixOS, not Home Manager.
